conda/Dockerfile

```diff
@@ -35,43 +38,45 @@ RUN /opt/conda/bin/conda install -y conda-package-handling=1.6.0
 
 # Install CUDA
 FROM base as cuda
+RUN rm -rf /usr/local/cuda-*
 ADD ./common/install_cuda.sh install_cuda.sh
 
-FROM cuda as cuda92
+FROM cuda as cuda9.2
 RUN bash ./install_cuda.sh 9.2
 
-FROM cuda as cuda101
+FROM cuda as cuda10.1
 RUN bash ./install_cuda.sh 10.1
 
-FROM cuda as cuda102
+FROM cuda as cuda10.2
 RUN bash ./install_cuda.sh 10.2
 
-FROM cuda as cuda110
+FROM cuda as cuda11.0
 RUN bash ./install_cuda.sh 11.0
 
-FROM cuda as cuda111
+FROM cuda as cuda11.1
 RUN bash ./install_cuda.sh 11.1
 
-FROM cuda as cuda112
+FROM cuda as cuda11.2
 RUN bash ./install_cuda.sh 11.2
 
 # Install MNIST test data
 FROM base as mnist
 ADD ./common/install_mnist.sh install_mnist.sh
 RUN bash ./install_mnist.sh
 
-FROM base as final
+FROM base as all_cuda
+COPY --from=cuda9.2   /usr/local/cuda-9.2  /usr/local/cuda-9.2
+COPY --from=cuda10.1  /usr/local/cuda-10.1 /usr/local/cuda-10.1
+COPY --from=cuda10.2  /usr/local/cuda-10.2 /usr/local/cuda-10.2
+COPY --from=cuda11.0  /usr/local/cuda-11.0 /usr/local/cuda-11.0
+COPY --from=cuda11.1  /usr/local/cuda-11.1 /usr/local/cuda-11.1
+COPY --from=cuda11.2  /usr/local/cuda-11.2 /usr/local/cuda-11.2
+
+FROM ${BASE_TARGET} as final
 COPY --from=patchelf /patchelf            /usr/local/bin/patchelf
 COPY --from=conda    /opt/conda           /opt/conda
-RUN rm -rf /usr/local/cuda-*
-COPY --from=cuda92   /usr/local/cuda-9.2  /usr/local/cuda-9.2
-COPY --from=cuda101  /usr/local/cuda-10.1 /usr/local/cuda-10.1
-COPY --from=cuda102  /usr/local/cuda-10.2 /usr/local/cuda-10.2
-COPY --from=cuda110  /usr/local/cuda-11.0 /usr/local/cuda-11.0
-COPY --from=cuda111  /usr/local/cuda-11.1 /usr/local/cuda-11.1
-COPY --from=cuda112  /usr/local/cuda-11.2 /usr/local/cuda-11.2
-ADD ./java/jni.h     /usr/local/include/jni.h
-ENV PATH /opt/conda/bin:$PATH
+ADD  ./java/jni.h    /usr/local/include/jni.h
+ENV  PATH /opt/conda/bin:$PATH
 COPY --from=mnist  /usr/local/mnist /usr/local/mnist
 RUN rm -rf /usr/local/cuda
```

Comment:
Why did we need to remove /usr/local/cuda before and do we need to do that after line 73 too?

Reply:
it's to remove the original `/usr/local/cuda` from the base `nvidia` image